### PR TITLE
[deploy test] Fix default series test

### PR DESCRIPTION
**Pull request by barrettj12**
A new revision 22 of the `ubuntu` charm was published, which adds storage to the charm (see [charm-ubuntu#35](https://github.com/juju-solutions/charm-ubuntu/pull/35)). This was causing some failures in our CI, specifically the default series deploy tests.

Firstly, we were destroying the model before the charm had time to reach a steady state. This seems to leave the storage in a broken state which blocks model teardown (see [lp#2015422](https://bugs.launchpad.net/juju/+bug/2015422)). To amend this, I've added a `wait_for` inside the test to allow the application to fully start before tearing things down. I've also added the `--storage "files=tmpfs"` flag which seems to set up the storage in a way that is easier to tear down.

Secondly, each subtest was testing both Charmhub and Charmstore deploys. Deploying the second charm was causing an error creating storage:
```
04:47:39 Located charm "ubuntu" in charm-hub, revision 22
04:47:39 Deploying "ubuntu" from charm-hub charm "ubuntu", revision 22 in channel stable on bionic
04:47:43 Located charm "ubuntu" in charm-store, revision 22
04:47:43 Deploying "csubuntu" from charm-store charm "ubuntu", revision 22 in channel stable on bionic
04:47:44 ERROR getting details for storage files/1: filesystem for storage instance "files/1" not found
```
To fix this, I've moved the Charmstore tests into separate subtest functions (`run_deploy_default_series_cs` and `run_deploy_not_default_series_cs`. This should isolate things more and also make it easier to merge through to 3.x (where these tests won't exist).

## QA steps

```sh
cd tests
./main.sh -v deploy test_deploy_default_series
```

### CODE REVIEW

File: tests/suites/deploy/deploy_default_series.sh [hunk 1]
```diff
@@ -7,14 +7,27 @@ run_deploy_default_series() {
...
+run_deploy_default_series_cs() {
+	echo
+
+	model_name="test-deploy-default-series-cs"
+	file="${TEST_DIR}/${model_name}.log"
+
+	ensure "${model_name}" "${file}"
+
+	juju model-config default-series=bionic
+	juju deploy cs:ubuntu --storage "files=tmpfs"
```
Comment (hpidcock) [inline tests/suites/deploy/deploy_default_series.sh:27]: does this actually use the latest ubuntu charm when coming from cs?

File: tests/suites/deploy/deploy_default_series.sh [hunk 2]
```diff
@@ -28,14 +41,27 @@ run_deploy_not_default_series() {
...
+run_deploy_not_default_series_cs() {
+	echo
+
+	model_name="test-deploy-not-default-series-cs"
+	file="${TEST_DIR}/${model_name}.log"
+
+	ensure "${model_name}" "${file}"
+
+	juju model-config default-series=bionic
+	juju deploy cs:ubuntu --series focal
```
Comment (hpidcock) [inline tests/suites/deploy/deploy_default_series.sh:61]: if the above question is answered and we need storage, should this not also have the `--storage` directive?

**Review by hpidcock — APPROVED** [review]
a couple of questions but LGTM thanks 😀

File: tests/suites/deploy/deploy_default_series.sh (hunk 1 above)
Comment (barrettj12) [reply tests/suites/deploy/deploy_default_series.sh:27]: Yep!

File: tests/suites/deploy/deploy_default_series.sh (hunk 2 above)
Comment (barrettj12) [reply tests/suites/deploy/deploy_default_series.sh:61]: thanks, missed one

File: tests/suites/deploy/deploy_default_series.sh (hunk 1 above)
Comment (hmlanigan) [reply tests/suites/deploy/deploy_default_series.sh:27]: @hpidcock cs is backed by charmhub via the compatibility api. If a charm has started life as cs, any revisions released to latest/stable in charmhub, is available via cs

**Pull request merged**